src/Ledger/Utxo/Properties.lagda

```diff
@@ -364,22 +418,56 @@ module DepositHelpers
        $ begin tot + (dep - ref) ≡˘⟨ +-∸-assoc tot ref≤dep ⟩
                (tot + dep) - ref ≡⟨ cong (_- ref) $ +-comm tot dep ⟩
                (dep + tot) - ref ≡˘⟨ deposits-change ⟩
-               uDep              ≡⟨ cong getCoin $ sym $ uDep≡ step ⟩
+               uDep              ≡⟨ cong getCoin $ sym h ⟩
                getCoin deposits' ∎ ⟩
     (fees + txfee) + getCoin deposits' + txdonation
       ∎
+
+  rearrange1 : cbalance (utxo ∣ collateral) + txdonation + getCoin deposits + fees
+               ≡ fees + cbalance (utxo ∣ collateral) + getCoin deposits + txdonation
+  rearrange1 =   begin
+    cbalance (utxo ∣ collateral) + txdonation + getCoin deposits + fees
+      ≡⟨ +-comm _ fees ⟩
+    fees ℕ.+ (cbalance (utxo ∣ collateral) ℕ.+ txdonation ℕ.+ getCoin deposits)
+      ≡t⟨⟩
+    (fees ℕ.+ cbalance (utxo ∣ collateral)) ℕ.+ (txdonation ℕ.+ getCoin deposits)
+      ≡⟨ cong ((fees + cbalance (utxo ∣ collateral)) +_) $ +-comm txdonation (getCoin deposits) ⟩
+    (fees + cbalance (utxo ∣ collateral)) ℕ.+ ((getCoin deposits) ℕ.+ txdonation)
+      ≡t⟨⟩
+    (fees + cbalance (utxo ∣ collateral)) ℕ.+ (getCoin deposits) ℕ.+ txdonation
+      ≡⟨ refl ⟩
+    (fees + cbalance (utxo ∣ collateral)) + getCoin deposits + txdonation
+      ∎
+
+  utxo-ref-prop' :
+    cbalance utxo ≡
+     (cbalance (utxo ∣ collateral ᶜ)
+            + cbalance (utxo ∣ collateral)) + txdonation
+  utxo-ref-prop' = begin
+    cbalance utxo
+      ≡˘⟨ balance-cong-coin {utxo = (utxo ∣ collateral ᶜ) ∪ˡ (utxo ∣ collateral)}{utxo}
+          $    disjoint-∪ˡ-∪ (disjoint-sym res-ex-disjoint)
+          ≡ᵉ-∘ ∪-sym
+          ≡ᵉ-∘ res-ex-∪ (_∈? collateral) ⟩
+    cbalance ((utxo ∣ collateral ᶜ) ∪ˡ (utxo ∣ collateral))
+      ≡⟨ (balance-∪ {utxo ∣ collateral ᶜ} {utxo ∣ collateral}
+         $ flip res-ex-disjoint) ⟩
+    (cbalance (utxo ∣ collateral ᶜ)) ℕ.+ cbalance (utxo ∣ collateral)
+      ≡⟨ {!!} ⟩
+    (cbalance (utxo ∣ collateral ᶜ) ℕ.+ (cbalance (utxo ∣ collateral))) ℕ.+ txdonation
```

Comment:
This only seems provable if `txdonation` is always 0, which is not the case. I'm not sure what @Ali-Hill has in mind to complete this proof.

Reply:
This is resolved by just not paying out donations in the `Scripts-No` case.